interpreter/cling/lib/Interpreter/LookupHelper.cpp

```diff
@@ -1211,8 +1195,11 @@ namespace cling {
             S.InstantiateFunctionDefinition(SourceLocation(), TheDecl,
                                             true /*recursive instantiation*/);
           }
-          if (TheDecl->isInvalidDecl()) {
-            // if the decl is invalid try to clean up
+          if (TheDecl->isInvalidDecl() &&
```

Comment:
in particular for this case (invalid decl but no errors ocured) not sure if it's more appropriate to just unload the decl, or propagate an error such that the cleanup is done by failing the transaction as for the case where an error occurred.

Reply:
It might be.  I am not sure.  If I recalled correctly, when that code was written the transaction and transaction unloading was not as mature as it is now.

Related I do not know whether this code path (the related path below) is exercised in the tests (to test it out, what could put an obvious message and a process termination here).  If it is not exercise we should probably add a test that does.

Reply:
Do we have an example of a declaration being marked as invalid while no diagnostic has been generated?